protocol_rfcs/collated-string-type.md

```diff
@@ -0,0 +1,185 @@
+      "type" : "map",
+      "keyType": "string"
+      "valueType": {
+        "type": "array"
+        "elementType": {
+          "type": "map"
+          "keyType: {
+            "type": "array"
+            "elementType": "string"
+          },
+          "valueType": {
+            "type": "map",
+            "keyType": "string",
+            "valueType": {
+              "type": "struct",
+              "fields": [ {
+                "name": "f1"
+                "type": "string"
+              } ],
+            },
+            "metadata": {
+              "__COLLATIONS": { "f1": "ICU.de_DE" }
+            }
+          }
+        }
+      }
+      "metadata": {
+        "__COLLATIONS": {
+          "col1.key": "ICU.en_US",
+          "col1.value.element.key": "ICU.en_US",
+          "col1.value.element.value.key": "ICU.en_US"
+        }
+      }
+    }
+  ]
+}
+```
+
+#### Collation versions
+
+The [Domain Metadata](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#domain-metadata) for the `collations` table feature contains hints which version of a collation should be used to read from the table and for which versions of a collations clients should produce statistics when writing to the table. They allow clients to choose a collation version without having to look at the statistics of all AddFiles first. Clients are allowed to ignore the hints.
+
+`collations` [Domain Metadata](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#domain-metadata)
+
+```
+{
+  "readVersions": {
+    "ICU.en_US": "73"
+  },
+  "writeVersions": {
+    "ICU.en_US": ["72", "73"]
+  }
+}
+```
+
+> ***Update the string row in the [Primitive Types](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#primitive-types) table.***
+
+### Primitive Types
+
+Type Name | Description
+-|-
+string| UTF-8 encoded string of characters. A collation can be specified in [Column Metadata](#specifying-collations-in-the-table-schema), otherwise binary collation is used as the default.
+
+> ***Add new rows to the [Column Metadata](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#column-metadata) table.***
+
+Field Name | Description
+-|-
+__COLLATIONS | Collations for strings stored in the field or combinations of maps and arrays that are stored in this field and do not have nested structs. Refer to [Specifying collations in the table schema](#specifying-collations-in-the-table-schema) for more details.
+
+> ***Edit the [Per-file Statistics](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#per-file-statistics) section and change it from the "Per-column statistics" section onwards.***
+
+Per-column statistics record information for each column in the file and they are encoded, mirroring the schema of the actual data.
+For example, given the following data schema:
+```
+|-- a: struct
+|    |-- b: struct
+|    |    |-- c: long
+|-- d: struct
+     |-- e: string collate ICU.en_US.72
+```
+
+Statistics could be stored with the following schema:
```

Comment:
Can the stats collection be optional for writers? For example a writer that doesn't understand collations or has no support for a particular collation, can write the regular stats and ignore the collation stats?